Makefile

```diff
@@ -27,10 +28,20 @@ test-skopeo:
 		skopeo_path=$${GOPATH}/src/github.com/projectatomic/skopeo && \
 		vendor_path=$${skopeo_path}/vendor/github.com/containers/image && \
 		git clone -b $(SKOPEO_BRANCH) https://github.com/$(SKOPEO_REPO) $${skopeo_path} && \
+		rm -fr $${skopeo_path}/vendor/github.com/Sirupsen/logrus && \
+		rm -fr $${skopeo_path}/vendor/github.com/docker/go-units && \
+		rm -fr $${skopeo_path}/vendor/golang.org/x/net && \
+		rm -fr $${skopeo_path}/vendor/github.com/vbatts/tar-split && \
+		rm -fr $${skopeo_path}/vendor/github.com/opencontainers/runc && \
+		mkdir -p $${skopeo_path}/vendor && \
+		ln -s . $${skopeo_path}/vendor/src && \
+		GOPATH=$${skopeo_path}/vendor go get -u $(BUILDFLAGS) github.com/containers/storage/storage && \
+		rm -f $${skopeo_path}/vendor/src && \
 		rm -rf $${vendor_path} && cp -r . $${vendor_path} && \
 		cd $${skopeo_path} && \
-		make binary-local test-all-local && \
-		$(SUDO) make check && \
+		sed -i '/WORKDIR/{h;s/.*/RUN dnf -y install btrfs-progs-devel device-mapper-devel/g;G}' Dockerfile && \
+		make binary-local test-all-local BUILDTAGS="btrfs_noversion libdm_no_deferred_remove" && \
+		$(SUDO) make check BUILDTAGS="btrfs_noversion libdm_no_deferred_remove" && \
```

Comment:
Sounds good.  I don't expect most PRs here to include a corresponding PR in skopeo, so we'd want CI to keep testing with skopeo's master branch.